lib/EarlyFunctionAnalysis/DetectABI.cpp

```diff
@@ -488,6 +507,81 @@ void DetectABI::interproceduralPropagation() {
   }
 }
 
+void DetectABI::propagatePrototypesInFunction(model::Function &Function) {
+  const MetaAddress &Entry = Function.Entry();
+
+  FunctionSummary &Summary = Oracle.getLocalFunction(Entry);
+  ABIAnalyses::ABIAnalysesResults &ABI = Summary.ABIResults;
+  SortedVector<efa::BasicBlock> &CFG = Summary.CFG;
+  std::set<llvm::GlobalVariable *> &WrittenRegisters = Summary.WrittenRegisters;
+
+  const bool HasArguments = ABI.ArgumentsRegisters.size() > 0;
+  const bool HasReturns = ABI.FinalReturnValuesRegisters.size() > 0;
+  const bool IsSingleNode = CFG.size() == 1;
+
+  if (HasReturns or HasArguments or !IsSingleNode) {
+    return;
+  }
+
+  efa::BasicBlock &Block = *CFG.begin();
+
+  const bool HasSingleSuccessor = Block.Successors().size() == 1;
+  if (!HasSingleSuccessor) {
+    return;
+  }
+
+  auto &Successor = *Block.Successors().begin();
+
+  // Select new prototype for wrapper function
+  if (const auto &Call = dyn_cast<efa::CallEdge>(Successor.get())) {
+    auto Prototype = getPrototype(*Binary, Entry, Block, *Call);
+
+    using abi::FunctionType::Layout;
+    // Get layout of wrapped function
+    Layout CalleeLayout = Layout::make(Prototype);
+
+    // Verify that wrapper function:
+    //  - don't write stack pointer
+    //  - don't write to callee arguments
+    //  - every store instruction writes to registers (not memory)
+    llvm::BasicBlock *BB = GCBI.getBlockAt(Block.ID().start());
```

Comment:
We have `using namespace llvm`, drop all `llvm::`.

Reply:
The problem is that we also have `efa::BasicBlock` and skipping namespace here generates a conflict. In other places I'll remove namespace.